### PR TITLE
Add fast approximate antialiasing (FXAA) to Viewport

### PR DESCRIPTION
**Pull request by Calinou**
This backports FXAA from the `master` branch.

### GLES3

image

### GLES2

image

### CODE REVIEW

File: drivers/gles2/shaders/tonemap.glsl [hunk 1]
```diff
@@ -138,7 +142,7 @@ uniform ivec2 glow_texture_size;
 vec4 texture2D_bicubic(sampler2D tex, vec2 uv, int p_lod) {
 	float lod = float(p_lod);
 	vec2 tex_size = vec2(glow_texture_size >> p_lod);
-	vec2 pixel_size = vec2(1.0) / tex_size;
+	vec2 pix_size = vec2(1.0) / tex_size;
```
Comment (lawnjelly) [inline drivers/gles2/shaders/tonemap.glsl:146]: Could we maybe use something like e.g. `pixel_size` and `inv_pixel_size`, or `texel_size` if this is specific to this routine?

Having `pixel_size` and `pix_size` doesn't make it clear what either is referring to without looking them up.

Comment (clayjohn) [reply drivers/gles2/shaders/tonemap.glsl:146]: I agree. texel_size would be a more appropriate name here. And then it is distinguished from pixel_size.

**Comment by capnm** [comment]
@Calinou why did you move the config from the environment to project settings?

**Comment by Calinou** [comment]
@capnm This was done at clayjohn's request for consistency with MSAA (and how settings will work in 4.0).

**Comment by capnm** [comment]
Now an project without a def. environment shows a black screen ...

**Comment by clayjohn** [comment]
@capnm probably just a missed if statement. Is that happening in GLES2 or GLES3?

**Comment by capnm** [comment]
:bug: is in GLES3.

<details>
<summary>Pictures</summary>


Bildschirmfoto von 2020-09-17 16-21-00
Bildschirmfoto von 2020-09-17 16-21-03

Bildschirmfoto von 2020-09-17 16-22-19
Bildschirmfoto von 2020-09-17 16-22-21




</details>

**Comment by clayjohn** [comment]
@capnm Thanks for the images! However, I can't reproduce. Have you merged this into a custom version of Godot? I see you have some project settings that I don't have.

**Comment by capnm** [comment]
> Have you merged this into a custom version of Godot?

Yes, but that shouldn't interfere here, I'll test it again with the current 3.2 tip.
After deactivating the default environment, you have to **restart the editor**!

> I see you have some project settings that I don't have.

They accumulate and sometimes get stuck in the UI from older 3.2 versions. I'll create new project...

**Comment by clayjohn** [comment]
> After deactivating the default environment, you have to restart the editor!

I didn't do that. I will try again tonight and see if I can reproduce! I'm sure it will be an easy fix. :)

**Comment by capnm** [comment]
> I didn't do that. I will try again tonight and see if I can reproduce! I'm sure it will be an easy fix. :)

Checked again 3.2 tip + this PR (Linux), removed all old Godot's caches and configs, same issue ;)
[test-FXAA2.zip](https://github.com/godotengine/godot/files/5240493/test-FXAA2.zip)
<details>
<summary>Pictures</summary>

Bildschirmfoto von 2020-09-17 18-07-45

</details>

**Comment by capnm** [comment]
> I'm sure it will be an easy fix. :)

In fact it was :+1: Thank you both.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**